scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,249 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+arg_names=""
+arg_kubeconfig=""
+arg_context=""
+arg_ip_addr=""
+arg_merge=false
+arg_output=""
+arg_verbose=true
+
+
+display_help() {
+  cat << EOF
+Usage: $0 [--kubeconfig <filename>] [--context <name>] [--names <list of names>] [--replace-local-ip-address <address>] [--merge] [-o <filename>] [-V] [-X]
+--kubeconfig <filename>                 use the specified kubeconfig
+--context <name>                        use the specified context
+--names <name1>,<name2>                 comma separated list of KubeFlex Control Planes names to import, instead of default *all*
+--replace-local-ip-address <address>    replaces server addresses "127.0.0.1" with <address>
+--merge                                 merge the control planes contexts with the existing cluster contexts
+-o <filename>                           specify a different kubeconfig file to save the contexts (- for stdout)
+--silent                                no information output
+-X                                      enable verbose execution of the script for debugging
+EOF
+}
+
+
+get_kubeconfig() {
+    context="$1"
+    cp_name="$2"
+    cp_type="$3"
+    ip_addr="$4"
+
+    echov "Getting the kubeconfig of Control Plane \"$cp_name\" of type \"$cp_type\" in context \"$context\":"
+
+    # wait for CP ready
+    while [[ $(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do
+        echov "* Waiting for Control Plane \"$cp_name\" in context \"$context\" to be Ready..."
+        sleep 5
+    done
+
+    # get out of cluster kubeconfig
+    if [[ "$cp_type" == "host" ]] ; then
+        echov "- Using cluster context \"${context}\""
+        kubeconfig=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context config view --flatten --minify)
+    else
+        # determine the secret name and namespace
+                    key=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.key}')
+            secret_name=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.name}')
+        secret_namespace=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.namespace}')
+        # get the kubeconfig
+        echov "- Using \"$key\" from \"$secret_name\" secret in \"$secret_namespace\""
+        kubeconfig=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}" | base64 -d)
+    fi
+
+    # make kubeconfig unique for the control plane
+    echov "- Making the kubeconfig unique..."
+    cluster=$(kubectl --kubeconfig <(echo "$kubeconfig") config get-clusters | tail -n +2)
```

Comment:
This method of passing the kubeconfig contents does not work for me on my Mac (MacOS 14.5, bash 3.2.57):

```
mspreitz@mjs13 kubestellar % (set -x; kubectl --kubeconfig <(cat ~/.kube/config) config current-context)
+-zsh:252> cat /Users/mspreitz/.kube/config
+-zsh:252> kubectl --kubeconfig /dev/fd/11 config current-context
error: current-context is not set
```

even though

```
mspreitz@mjs13 kubestellar % kubectl --kubeconfig ~/.kube/config config current-context
kind-kubeflex
```

Reply:
bash too:

```
bash-3.2$ kubectl --kubeconfig <(cat ~/.kube/config ) config current-context
error: current-context is not set
bash-3.2$ kubectl --kubeconfig ~/.kube/config config current-context
kind-hub
bash-3.2$ echo $0
bash
bash-3.2$ bash --version
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin23)
Copyright (C) 2007 Free Software Foundation, Inc.
```

Reply:
@MikeSpreitzer could this be related to the version of `kubectl`?

```
$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.27.4
Kustomize Version: v5.0.1
Server Version: v1.27.2+k3s1

$ (set -x; kubectl --kubeconfig <(cat ~/.kube/config) config current-context)
+ kubectl --kubeconfig /dev/fd/63 config current-context
++ cat /home/vagrant/.kube/config
its1
```

Reply:
My kubectl version is 1.27.2, close to yours.

Perhaps more important is the my next comment, these lines are assigning to variables that are not used (so could be deleted).

Reply:
ok removed